File: killrvideo-service-comments/src/main/java/com/killrvideo/service/comment/dto/QueryCommentByVideo.java
package com.killrvideo.service.comment.dto;

import com.killrvideo.dse.dto.QueryDefinition;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Optional;
import java.util.UUID;

/**
 * Query to search comments for a User.
 *
 * @author DataStax Developer Advocates team.
 */
@Getter
@Setter
@NoArgsConstructor
public class QueryCommentByVideo extends QueryDefinition {
    private static final long serialVersionUID = 7721676513515347779L;

    /**
     * Video unique identifier.
     */
    private UUID videoId = null;

    /**
     * Comment offset if specified (Optional)
     */
    private Optional<UUID> commentId = Optional.empty();
}

File: killrvideo-service-comments/src/main/kotlin/com/killrvideo/service/comment/dto/QueryCommentByVideo.kt
package com.killrvideo.service.comment.dto

import java.util.*

/**
 * Query to search comments for a User.
 *
 * @author DataStax Developer Advocates team.
 */
data class QueryCommentByVideo(
    val videoId: UUID,
    val commentId: UUID?,
    val pageSize: Int,
    val pageState: String?
)
